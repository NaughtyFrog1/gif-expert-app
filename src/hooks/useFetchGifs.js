import { useEffect, useState } from 'react'
import getGifs from '../helpers/getGifs'

const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  })

  useEffect(() => {
    getGifs(category).then((gifs) => {
      setTimeout(() => {
        setState({
          data: gifs,
          loading: false,
        })
      }, 2000);
    })
  }, [category])

  return state
}

export default useFetchGifs

/**
 * Los custom hooks funcionan como functional components, pero no retornan un JSX
 */
