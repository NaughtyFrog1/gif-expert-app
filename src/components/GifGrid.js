import React from 'react'
import PropTypes from 'prop-types'
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category)

  return (
    <div className="category">
      <h3>{category}</h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}

      <ol className="gif-cards">
        {gifs.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </ol>
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifGrid
