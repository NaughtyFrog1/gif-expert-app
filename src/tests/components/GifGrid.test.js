import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import useFetchGifs from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('test for GifGrid', () => {
  test('should render the component properly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    })

    const wrapper = shallow(<GifGrid category="The office" />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should show items when images are loaded', () => {
    useFetchGifs.mockReturnValue({
      data: [{ id: 'ABC', url: 'https://foo/bar.gif', title: 'FooBar' }],
      loading: false,
    })

    const wrapper = shallow(<GifGrid category="The office" />)
    expect(wrapper).toMatchSnapshot()
  })
})
