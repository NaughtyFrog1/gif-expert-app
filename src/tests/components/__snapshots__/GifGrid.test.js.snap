// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for GifGrid should render the component properly 1`] = `
<div
  className="category"
>
  <h3>
    The office
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <ol
    className="gif-cards"
  />
</div>
`;

exports[`test for GifGrid should show items when images are loaded 1`] = `
<div
  className="category"
>
  <h3>
    The office
  </h3>
  <ol
    className="gif-cards"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="FooBar"
      url="https://foo/bar.gif"
    />
  </ol>
</div>
`;
